Add blob lease to existing operations

BrianMcBrayer opened the pull request:
* Added lease parameter to all existing blob operations
* Added unit tests to cover the lease cases for those operations
* Cruft removal

tarwn commented:
Review Notes

* Test Naming - Several of your tests only have two part naming, missing the underscore between situation and expectation: "PutBlockAsync_LeasedBlobWithIncorrectLeaseGivenThrowsSpecifiedException" should be "PutBlockAsync_LeasedBlobWithIncorrectLeaseGiven_ThrowsSpecifiedException"
* It's not a big enough deal to change, but I don't like the TestCases for this situation. They optimize for the first developer to write the code at the expense of the one that will get a more generic message and have to go read the code to understand what went wrong. One case is testing a local guard, two cases are testing that the error handling returns the expected result after a full integration round-trip. The "null" case isn't even testing what the name advertises, it is actually testin something I would call "PutBlockAsync_LeasedBlobAndNoLeaseSpecified_ThrowsLeaseIdMissingAzureException".
* "const string" variable declaration in a couple tests, versus "var" everywhere else?
* Several tests look like they are spaced out in to AAA sections, but don't actually have asserts, example:  GetBlob_LeasedBlobWithoutLeaseSpecified_GetsBlobWithoutException - either comment the "Arrange" and "Act" sections or adjust the spacing to make them obvious at a glance
* I still don't like the test cases for exception checks - in fact I just realized that for the guard case, we're doing all of the Azure container and blob work for a test case that (if it runs successfully) will not even use those resources, so those tests could have been 3 lines long:

BrianMcBrayer commented:
Updated per your directions. Removed the test cases (it didn't take long); updated the vars for consistency.
I use 'consts' when I can to specify that I don't expect the values to change. I only use vars when I have to.
